File: src/main/java/iot/tetracubered/properties/SmartHubProperties.java
package iot.tetracubered.properties;

import org.eclipse.microprofile.config.inject.ConfigProperty;

import io.quarkus.arc.config.ConfigProperties;

@ConfigProperties(prefix = "smart-hub")
public interface SmartHubProperties {

    @ConfigProperty(name = "topics")
    TopicsProperties topics();
}

File: src/main/kotlin/red/tetracube/configurations/properties/SmartHubProperties.kt
package red.tetracube.configurations.properties

import io.quarkus.arc.config.ConfigProperties
import org.eclipse.microprofile.config.inject.ConfigProperty

@ConfigProperties(prefix = "smart-hub")
interface SmartHubProperties {

    @ConfigProperty(name = "iot")
    fun iot(): IoTProperties
}